### DISCUSSION TITLE
ESP32 appears to be working, but noting on the front panel is working.

### DISCUSSION THREAD

**Discussion by EPSavage** [thehighnibble/thehighnibble.github.io#157] [IMSAI8080esp replica]
The on/off sw does nothing. I can't get the IMSAI into a RUN state. The log seems to indicate all is well.
Here is the log.

=~=~=~=~=~=~=~=~=~=~=~= PuTTY log 2024.07.06 17:36:15 =~=~=~=~=~=~=~=~=~=~=~=
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x3f (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4940
ho 0 tail 12 room 4
load:0x40078000,len:15120
load:0x40080400,len:3764
entry 0x40080620
[0;32mI (306) psram: This chip is ESP32-PICO[0m
[0;32mI (306) spiram: Found 64MBit SPI RAM device[0m
[0;32mI (306) spiram: SPI RAM mode: flash 80m sram 80m[0m
[0;32mI (309) spiram: PSRAM initialized, cache is in low/high (2-core) mode.[0m
[0;32mI (316) cpu_start: Pro cpu up.[0m
[0;32mI (320) cpu_start: Starting app cpu, entry point is 0x40081c40[0m
[0;32mI (0) cpu_start: App cpu up.[0m
[0;32mI (811) spiram: SPI SRAM memory test OK[0m
[0;32mI (854) cpu_start: Pro cpu start user code[0m
[0;32mI (854) cpu_start: cpu freq: 240000000[0m
[0;32mI (854) cpu_start: Application information:[0m
[0;32mI (856) cpu_start: Project name:     imsaisim_esp32[0m
[0;32mI (862) cpu_start: App version:      v1.12.0[0m
[0;32mI (867) cpu_start: Compile time:     Jun 10 2023 03:02:39[0m
[0;32mI (873) cpu_start: ELF file SHA256:  cd219fa8e9f544b1...[0m
[0;32mI (879) cpu_start: ESP-IDF:          v4.4.4-1-gadda465dc3[0m
[0;32mI (885) heap_init: Initializing. RAM available for dynamic allocation:[0m
[0;32mI (892) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM[0m
[0;32mI (898) heap_init: At 3FFBBD00 len 00024300 (144 KiB): DRAM[0m
[0;32mI (905) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM[0m
[0;32mI (911) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM[0m
[0;32mI (917) heap_init: At 4009DAF0 len 00002510 (9 KiB): IRAM[0m
[0;32mI (923) spiram: Adding pool of 3629K of external SPI memory to heap allocator[0m
[0;32mI (932) spi_flash: detected chip: gd[0m
[0;32mI (936) spi_flash: flash io: qio[0m
[0;32mI (940) cpu_start: Starting scheduler on PRO CPU.[0m
[0;32mI (0) cpu_start: Starting scheduler on APP CPU.[0m
[0;31mE (990) nvs: Error (ESP_ERR_NVS_NOT_FOUND) opening NVS handle![0m
[0;33mW (990) nvs: Defaulting log level NVS_LOG_LEVEL = INFO[0m
[0;32mI (992) esp32_boot: Log Level set to INFO[0m
[0;32mI (996) esp32_hardware: Shell Running on Core#0[0m
[0;32mI (1482) esp32_hardware: CP-A NOT Found - Power Switch 0[0m
*************************************************************************
* CONGRATULATIONS & WELCOME TO THE IMSAI 8080esp
*
* You can disregard any earlier NVS_NOT_FOUND error for now.
*   - you will configure the NVS later, refer to the Configuration guide.
*
* Seeing this message suggests that you've soldered the PSRAM (SPI SRAM)
* successfully and it is passing its tests, see above.
*
* Check the uSD tests below to know if that has also been soldered
* successfully.
*
* (If this is not a new IMSAI 8080esp then the PWR switch is faulty.)
*************************************************************************
[0;32mI (1532) esp32_hardware: DIP/MEM Switches: 00[0m
[0;32mI (1537) esp32_hardware: Initializing uSD card[0m
[0;32mI (1542) gpio: GPIO[13]| InputEn: 0| OutputEn: 1| OpenDrain: 0| Pullup: 0| Pulldown: 0| Intr:0 [0m
[0;32mI (1605) esp32_hardware: uSD card successfully detected and mounted with the following details:[0m
Name: SDABC
Type: SDHC/SDXC
Speed: 40 MHz
Size: 15203MB
*************************************************************************
* The uSD tests appear to have succeeded.
*
* NOTE: THIS COULD BE A FALSE POSITIVE IF THE ESP32 HAS RESTARTED
*
* Please check all earlier logs and/or press a H/W reset switch and
* check the logs. Any error with the uSD is a bad sign,
* even if it can succeed after a restart.
* If any error with the uSD is observed, check all soldering and the
* RN1 resistor network, including the orientation of RN1.
*
*************************************************************************
[0;32mI (1658) esp32_update: Checking for image update package[0m
[0;32mI (1665) esp32_boot: OPENING BOOT.CONF[0m
[0;32mI (1670) esp32_boot: OPENED: /sdcard/imsai/conf/boot.conf[0m
[0;32mI (1676) esp32_boot: ENV: NTP_SERVER=pool.ntp.org[0m
[0;32mI (1680) esp32_boot: ENV: TZ=AEST-10AEDT,M10.1.0,M4.1.0[0m
[0;32mI (1686) esp32_boot: ENV: HOSTNAME=imsai8080[0m
[0;32mI (1691) esp32_boot: ENV: PORT=80[0m
[0;32mI (1695) esp32_boot: ENV: SSID=mySSID[0m
[0;32mI (1699) esp32_boot: ENV: PASSWORD=myPASSWORD[0m
[0;32mI (1704) esp32_boot: ENV: WIFI.sta.scan=1[0m
[0;32mI (1709) esp32_boot: ENV: TTY.netsrv.buffer_delay=33[0m
[0;32mI (1715) esp32_boot: ENV: LPT.netsrv.buffer_delay=33[0m
[0;32mI (1720) esp32_boot: ENV: HARDDISK=hd-ws4[0m
[0;32mI (1725) esp32_boot: ENV: MODEM.init=ATS15=1 [0m
[0;32mI (1730) esp32_hardware: Initialise UART1 to default configuration[0m
I (1739) wifi:wifi driver task: 3ffc3ea8, prio:23, stack:3584, core=0
[0;32mI (1743) system_api: Base MAC address is not set[0m
[0;32mI (1748) system_api: read default base MAC address from EFUSE[0m
I (1759) wifi:wifi firmware version: 6567a16
I (1760) wifi:wifi certification version: v7.0
I (1763) wifi:config NVS flash: enabled
I (1766) wifi:config nano formating: disabled
I (1771) wifi:Init data frame dynamic rx buffer num: 32
I (1775) wifi:Init management frame dynamic rx buffer num: 32
I (1781) wifi:Init management short buffer num: 32
I (1785) wifi:Init dynamic tx buffer num: 32
I (1790) wifi:Init tx cache buffer num: 32
I (1793) wifi:Init static rx buffer size: 1600
I (1798) wifi:Init static rx buffer num: 16
I (1801) wifi:Init dynamic rx buffer num: 32
[0;32mI (1806) wifi_init: rx ba win: 16[0m
[0;32mI (1809) wifi_init: tcpip mbox: 64[0m
[0;32mI (1813) wifi_init: udp mbox: 6[0m
[0;32mI (1817) wifi_init: tcp mbox: 10[0m
[0;32mI (1821) wifi_init: tcp tx win: 11680[0m
[0;32mI (1825) wifi_init: tcp rx win: 11680[0m
[0;32mI (1830) wifi_init: tcp mss: 1460[0m
[0;32mI (1834) wifi_init: WiFi IRAM OP enabled[0m
[0;32mI (1838) wifi_init: WiFi RX IRAM OP enabled[0m
[0;32mI (1843) wifi_init: LWIP IRAM OP enabled[0m
I (1849) wifi:Set ps type: 0

[0;32mI (1851) esp32_hardware: MAC ADDR: 30:83:98:df:c1:ed[0m
[0;32mI (1856) esp32_netsrv: AP Mode - Using SSID: imsai8080 (HOSTNAME)[0m
[0;32mI (1866) esp32_netsrv: Starting Wi-Fi now[0m
[0;32mI (1868) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07[0m
I (1958) wifi:mode : softAP (30:83:98:df:c1:ed)
I (1960) wifi:Total power save buffer number: 16
I (1960) wifi:Init max length of beacon: 752/752
I (1960) wifi:Init max length of beacon: 752/752
[0;32mI (1966) esp32_netsrv: EVENT: WIFI_EVENT - 12[0m
[0;32mI (1970) esp32_netsrv: soft-AP started[0m
[0;32mI (1974) esp32_netsrv: Initializing hostname to: imsai8080[0m
[0;32mI (1980) dns_server: STARTING DNS SERVER[0m
[0;32mI (1985) start_dns_server: dns server task started[0m
[0;32mI (1990) dns_server: Socket created[0m
[0;32mI (1994) dns_server: Socket bound, port 53[0m
[0;32mI (1999) dns_server: Waiting for data[0m
[0;32mI (2990) esp32_netsrv: Initializing MDNS[0m
[0;32mI (3001) esp32_netsrv: Initializing SNTP[0m
[0;32mI (3001) netsrv: Using HTTP Port: 80 [80][0m
[0;32mI (3001) netsrv: Created Semaphore[0m
[0;32mI (3006) esp32_netsrv: WiFi result is 1[0m
[0;33mW (3008) esp32_boot: Cannot set system time by NTP in AP mode, or no NTP_SERVER set[0m
[0;32mI (3016) esp32_boot: The current date/time is: Thu Jan  1 11:00:02 1970 [2][0m
[0;32mI (3023) esp32_hardware: POST flash timer stopped[0m
[0;32mI (3029) esp32_boot: Starting Watchdog Timer[0m

**Comment by udo-munk** [comment]
Here the log says:
esp32_hardware: CP-A NOT Found - Power Switch 0
Looks like the esp32 has no access to the CP-A switches.

**Reply by kernelpanic19** [reply]
Thanks for this - I was confused by the panel LEDs intermittently not lighting during build testing (pre switch soldering), and this made me realize the power switch must be properly seated for detection during boot up.

**Comment by nbreeden2** [comment]
Did you solder the extra 6 pins to the EXP32?

**Comment by EPSavage** [comment]
The resolution: I had U6 and U9 swapped. Putting the correct chip into its socket corrected the issue.
